homing_missiles.py

```diff
@@ -109,12 +109,22 @@ def draw_entities(self, screen):
         pygame.draw.rect(screen, self.main_character_color_, main_character_rect)
         self.draw_friendly_projectiles(screen)
 
+    def determine_enemy_color(self, enemy):
+        enemy_type = enemy.get_type()
+        if enemy_type == "BasicGrunt":
+            return YELLOW
+        elif enemy_type == "NotSoBasicGrunt":
+            return WHITE
+
+        return RED
+
     def draw_target_tags(self, screen):
         for enemy in self.universe_.enemies():
+            enemy_color = self.determine_enemy_color(enemy)
```

Comment:
Yes, I noticed we are doing this twice too and we have to split up the logic for things. That being said, the "Targetting System" Should be a class that we inherit right? At the moment this is not the case so I had no choice but to put it in both "Targetting Systems". 

All the draw functions inside are also basically repeats too